src/biokbase/narrative/tests/test_auth.py

```diff
@@ -0,0 +1,198 @@
+import json
+import os
+
+import pytest
+from requests import HTTPError
+
+from biokbase.auth import (
+    TokenInfo,
+    get_agent_token,
+    get_auth_token,
+    get_display_names,
+    get_token_info,
+    init_session_env,
+    set_environ_token,
+    validate_token,
+)
+from biokbase.narrative.common.url_config import URLS
+from biokbase.narrative.common.util import kbase_env
+
+AUTH_URL = URLS.auth + "/api/V2/"
+
+unauth_error = {
```

Comment:
Have you seen vcr.py? Last time I was devPoC, I was thinking about switching over the EE2 tests to using vcr.py so that you could (potentially) test against real server responses, but there would be an issue with some job states. For auth, though, it could work well.

Reply:
Oh, I remember your presentation (or a couple demo slides?) on that! Yeah, that's a good idea. I'll check it out.

Reply:
On some reflection, I think, in this case since we're dealing with results of auth calls, I feel kinda weird about recording that kind of account information in the repo, even if it's just a test account. If we were talking about things like fetching objects or narratives or app specs, sure, but these are small enough auth info objects that are easy enough to mock.

That said, I do like vcr.py and think we should use it for handling app spec testing!